### ISSUE TITLE
Error installing `1.50-x86_64-unknown-linux-musl` toolchain on macOS

### ISSUE THREAD

**Issue by probablykasper** [rust-lang/rust#83154]
Installing the `<version>-x86_64-unknown-linux-musl` toolchain worked for Rust 1.49, but doesn't in Rust 1.50. I'm on x86_64 macOS 10.14.6.

Logs from running `rustup toolchain install 1.50-x86_64-unknown-linux-musl`:
```
warning: toolchain '1.50-x86_64-unknown-linux-musl' may not be able to run on this system.
warning: If you meant to build software to target that platform, perhaps try `rustup target add x86_64-unknown-linux-musl` instead?
info: syncing channel updates for '1.50-x86_64-unknown-linux-musl'
info: latest update on 2021-02-11, rust version 1.50.0 (cb75ad5db 2021-02-10)
info: downloading component 'cargo'
info: downloading component 'clippy'
info: downloading component 'rust-docs'
info: downloading component 'rust-std'
info: downloading component 'rustc'
info: downloading component 'rustfmt'
info: installing component 'cargo'
info: using up to 500.0 MiB of RAM to unpack components
info: installing component 'clippy'
info: installing component 'rust-docs'
 14.7 MiB /  14.7 MiB (100 %)   9.2 MiB/s in  1s ETA:  0s
info: rolling back changes
error: failed to install component: 'rust-docs-x86_64-unknown-linux-musl', detected conflict: '"share/doc/rust/html/std/keyword.self.html"'
```

Not 100% sure if this is the right repo for it, please let me know if not.

**Comment by probablykasper** [comment]
This probably should go in https://github.com/rust-lang/rustup

**Comment by ehuss** [comment]
Although it is a bit unfortunate that rustup fails to install, I think the underlying problem of rustdoc creating files that only differ in case is tracked in #25879 and this particular case is #80504.

**Comment by probablykasper** [comment]
@ehuss Any idea why this specific issue started in 1.50, and why it's not there in `1.50-x86_64-apple-darwin` or `1.50-x86_64-pc-windows-gnu`?

**Comment by ehuss** [comment]
It doesn't apply to the `x86_64-apple-darwin` targets because those are built on macOS with a case-insensitive filesystem, so the files get stomped on during build time and only one appears in the distribution.

I don't think there is any difference between `gnu` and `musl`, they both fail for me.

As for why it changed in 1.50, the way the distribution manifest is created changed as part of #79788 (I think, it changed around the day that merged), so that instead of just indicating there is a single directory to install, the manifest now lists every file individually.

cc @kinnison 
cc @pietroalbini Pietro: I'm not sure if it was intended that the component manifest for rust-docs changed from listing a single directory (`dir:share/doc/rust/html`) to listing each file individually.  I'm not sure it matters much either way, but I'm not too familiar with how the manifest is used.

**Comment by probablykasper** [comment]
@ehuss Ah got it, I'm guessing Windows is built with a case-insensitive filesystem as well then. Is there a good reason Linux isn't?

**Comment by kinnison** [comment]
@probablykasper Linux by default does not use case-insensitive filesystems.

**Comment by probablykasper** [comment]
Seems to be fixed by #83678
